integrations/os/os.go

```
@@ -0,0 +1,98 @@
+package os
+
+import (
+	"context"
+	"errors"
+	"os/exec"
+	"runtime"
+	"time"
+
+	"go.autokitteh.dev/autokitteh/internal/backend/fixtures"
+	"go.autokitteh.dev/autokitteh/internal/kittehs"
+	"go.autokitteh.dev/autokitteh/sdk/sdkexecutor"
+	"go.autokitteh.dev/autokitteh/sdk/sdkmodule"
+	"go.autokitteh.dev/autokitteh/sdk/sdktypes"
+)
+
+var ExecutorID = sdktypes.NewExecutorID(fixtures.NewBuiltinIntegrationID("os"))
+
+func New() sdkexecutor.Executor {
+	return fixtures.NewBuiltinExecutor(
+		ExecutorID,
+		sdkmodule.ExportFunction("command", command, sdkmodule.WithArgs("cmd", "*args")),
+		sdkmodule.ExportFunction("shell", shell, sdkmodule.WithArgs("cmd", "sh?")),
+	)
+}
+
+func execute(ctx context.Context, name string, args ...string) (sdktypes.Value, error) {
+	cmd := exec.CommandContext(ctx, name, args...)
+	cmd.WaitDelay = 250 * time.Millisecond // without this, cancellations do not work properly.
+	out, err := cmd.CombinedOutput()
+
+	var rc int
+
+	var exit *exec.ExitError
+	if errors.As(err, &exit) {
+		rc = exit.ExitCode()
+
+		if rc < 0 && (errors.Is(ctx.Err(), context.DeadlineExceeded) || errors.Is(ctx.Err(), context.Canceled)) {
+			// rc could not be determined and the context was canceled or timed out.
+			err = ctx.Err()
+		} else {
+			// rc could be determined or no error context occured.
+			err = nil
```

You mask the error returned from `cmd.CombinedOutput`, why?
If the process exited normally, then the returned error from `cmd.CombinedOutput` will be `nil`.

Also, do we want to account for very large output? Maybe use `io.LimitReader` on `cmd.Stdout`?

---

i don't really mask it. rc is determined and that's what is returned. the rc variable will appear in the result.